src/utils/workspaces.ts

```diff
@@ -32,6 +34,12 @@ function matchesPatterns(
 	}
 }
 
+function matchesDependency(match: Workspace, depVersion: string): boolean {
+	const versionMatch = semver.satisfies(match.config.version, depVersion)
+	const localDep = localPrefixes.some(prefix => depVersion.startsWith(prefix))
+	return versionMatch || localDep
+}
```

Comment:
I think I'd like to make this a little bit more strict and also check if its a version string before running it through `semver.satisfies`

```tsx
let LOCAL_DEPENDENCY_REGEX = /^(?:file|workspace):(.+)$/

function matchesDependency(
  workspace: Workspace,
  match: Workspace,
  depVersion: string,
  rootDir: string
): boolean {
  let localDepMatch = depVersion.match(LOCAL_DEPENDENCY_REGEX)
  if (localDepMatch) {
    let dependencyPath = path.resolve(workspace.dir, localDepMatch[0])
    let matchPath = path.resolve(rootDir, match.dir)
    return dependencyPath === matchPath
  } else {
    return semver.satisfies(match.config.version, depVersion)
  }
}
```